src/query/expression/src/types.rs

```diff
@@ -344,6 +346,54 @@ pub trait ValueType: Debug + Clone + PartialEq + Sized + 'static {
     fn column_memory_size(col: &Self::Column) -> usize {
         Self::column_len(col) * std::mem::size_of::<Self::Scalar>()
     }
+
+    // Compare two scalars and return the Ordering between them.
+    #[inline(always)]
+    fn compare(_: Self::ScalarRef<'_>, _: Self::ScalarRef<'_>) -> Ordering {
+        unreachable!()
```

Comment:
it's dangerous to use `unreachable` , just use : 

```
fn compare(a: Self::ScalarRef<'_>, b: Self::ScalarRef<'_>) -> Ordering {
      a.partial_cmp(b).unwrap_or(Ordering::Equal)
}

since: type Scalar: Debug + Clone + PartialEq;
```